### PR TITLE
Rhfogh develop

### PR DESCRIPTION
**Pull request by rhfogh**
Added docker with installation mechanisms using commercial PyQt version, and tests for using licensed versions only.

NB the install machinery is a hack - the mxcubecore requirements are in requirements_mxcubecore.txt instead of using the mxcubecore pyproject.toml. This is undesirable, but I could not get the docker installation to work in any other way. Also, there might be a better location of these files than the top level of mxcubeqt

I would *very much* welcome if someone could check that the docker making works and ideally tell me of a better way to make the docker and handle the installation

The way to run the docker is to make it using the dockerfile, and then run start_mxcube, after editing to adapt to local directlry structure.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# RUN pip install poetry
+RUN pip install devtools
+
+## mxcubeqt requirements
+WORKDIR /tmp
+COPY ./PyQt5_commercial-5.15.9-cp37-abi3-manylinux_2_17_x86_64.whl /tmp
+RUN pip install ./PyQt5_commercial-5.15.9-cp37-abi3-manylinux_2_17_x86_64.whl
+RUN apt-get install -y python3-pyqtgraph
+RUN mkdir -p /tmp/mxcubecore
+COPY ./requirements_mxcubecore.txt ./setup.py /tmp/mxcubecore/
```
Comment (marcus-oscarsson) [inline Dockerfile:32]: Not sure I understand this bit, are you copying the `mxcubeqt` version of `requirements_mxcubecore` and `setup.py `or where does these files come from.

**Comment by marcus-oscarsson** [comment]
Looks good to me in general, thanks ! 

_For the `dockerfile` I cant see from where you are getting mxcubecore ?_

File: requirements_python3.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 mxcubecore
```
Comment (marcus-oscarsson) [inline requirements_python3.txt:1]: I guess this should be enough to install the requirements in [requirements_mxcubecore.txt](https://github.com/mxcube/mxcubeqt/pull/441/files#diff-50af65468a5b0ee24ca9b2d0991fd4a3ce58bae186aa8a8db79642888fced75a)

Comment (marcus-oscarsson) [reply requirements_python3.txt:1]: Id also suggest simply naming the file requirements.txt (as Python 3 is given now)

File: Dockerfile (hunk 1 above)
Comment (marcus-oscarsson) [reply Dockerfile:32]: I miss the bit where you "install" the mxcubeqt module i.e use `requirements_python3.txt`

**Comment by rhfogh** [comment]
"_For the dockerfile I cant see from where you are getting mxcubecore_ ?"
Actually I am not. in start_mxcube the pythonpath is set to $MXCUBE_BASE/mxcubeqt:$MXCUBE_BASE/mxcubecore
and MXCUBE_BASE is set also in start_mxcube and mounted on the docker. The effect is that mxcubeqt and mxcubecore are assumed to eb in parallel directories, and the directory containing them is passed in and mounted. 
The point is that I do not want to have mxcubecore as a subdirectory of mxcubeqt, since they are separate repositories adn in development and it all gets messy. This was the way I found to get it working without that.

**Comment by marcus-oscarsson** [comment]
Ok, simply running `poetry install` in the `mxcubecore` directory should work then I think. Otherwise those dependencies should come with the installation of `mxcubecore` in `requirements_python3.txt` right ?

**Comment by rhfogh** [comment]
"_I miss the bit where you "install" the mxcubeqt module i.e use requirements_python3.txt_"

Again, I actually do not. requirements_python2.txt and requirements_python2.txt are not used at all, currently - I just did not want to delete them before it was clear what we wanted to do. But then they are not needed, currently. pyqtgraph is installed in the Dockerfile, and  mxcubecore is handled as a separate repository and not installed via pip, so I know exactly what I am getting and do not need to go via an install step. And apart from pyqtgraph and mxcubecore there is nothing in those requirement files.
setup.py and requirements_mxcubecore.txt I added myself, and the latter contains all the requirements that are otherwise in teh mxcubecore pyproject.toml, since the latter is not used..

I guess you see why I wanted to discuss how this can be improved or reconciled with the normal way of installation?

**Comment by rhfogh** [comment]
"_Ok, simply running poetry install in the mxcubecore directory should work then I think_"

It did not work for me. I got either error message about modules not being defined, or I got problems with the fact that you can only access subdirectories of the directory containing the docker code. Anyway I could not find a way to do it. And I would really prefer not to have mxcubecore as a subdirectory of mxcubeqt while doing development work.

**Comment by rhfogh** [comment]
Maybe I should explain what I am trying to do here - in part because maybe someone can tell me I ought to be doing something else ;-)

This docker is intended for development, not distribution. I want to be able to keep editing the MXCUBE code without having to reinstall or remake the docker. For that reason, mxcubeqt, mxcubecore and the GPhL code are kept *outside* the docker, in separate repositories, and only the code dependencies are installed in the docker. As a separate point I wanted to avoid having mxcubecore as a subdirectory under mxcubeqt, since it can get quite messy with git repositories inside other git repositories. 

But I was worried that this would not fit in the normal approach particularly well.

**Comment by marcus-oscarsson** [comment]
So Ill merge this as it is in the mean time as we agreed on the meeting we had.

**Pull request merged**